lib/active_merchant/billing/gateways/jackson_river.rb

```diff
@@ -62,14 +62,74 @@ def add_address(post, options)
       end
 
       def add_invoice(post, money, options)
-        post[:amount] = amount(money).to_s # yup Jackson River wants it as a string :(
         post[:currency] = (options[:currency] || currency(money))
-        post[:recurs_monthly] = options[:recurring] if options[:recurring]
-          .present?
-        post[:recurring_other_amount] = amount(money).to_s if options[
-          :recurring
-        ]
-        post[:recurring_amount] = 'other' if options[:recurring]
+
+        # number OR string
+        #
+        # The value of this field may be a number or the word "other" if
+        #  an Other Amount value is being passed
+        post[:amount] = options[:amount] if options[:amount].present?
+
+        # number
+        #
+        # Other value for one-time donations
+        if options[:other_amount].present?
+          post[:other_amount] = options[:other_amount]
+        end
+
+        # string
+        #
+        # Allowed values are NO_RECURR, recurs, sb_fs_quarterly, sb_fs_semi, sb_fs_annually.
+        #
+        # NO_RECURR must be paired with amount and other_amount;
+        # the others must be paired with their corresponding amount and recurring_amount fields
+        if options[:recurs_monthly].present?
+          post[:recurs_monthly] = options[:recurs_monthly]
+          add_frequencies_amount(post, options)
+        end
+
+        # number  OR string
+        #
+        # The value of this field may be a number or the word "other" if
+        #  an other_amount value is also being passed
+        if options[:recurring_amount].present?
+          post[:recurring_amount] = options[:recurring_amount]
+        end
+
+        # number
+        #
+        # Other value for monthly recurring donations
+        if options[:recurring_other_amount].present?
+          post[:recurring_other_amount] = options[:recurring_other_amount]
+        end
+      end
+
+      def add_frequencies_amount(post, options)
+        if options[:sb_fs_quarterly_amount].present?
+          post[:sb_fs_quarterly_amount] = options[:sb_fs_quarterly_amount]
+        end
+
+        if options[:sb_fs_quarterly_other_amount].present?
+          post[:sb_fs_quarterly_other_amount] =
+            options[:sb_fs_quarterly_other_amount]
+        end
+
+        if options[:sb_fs_semi_amount].present?
+          post[:sb_fs_semi_amount] = options[:sb_fs_semi_amount]
+        end
+
+        if options[:sb_fs_semi_other_amount].present?
+          post[:sb_fs_semi_other_amount] = options[:sb_fs_semi_other_amount]
+        end
+
+        if options[:sb_fs_annually_amount].present?
+          post[:sb_fs_annually_amount] = options[:sb_fs_annually_amount]
+        end
+
+        if options[:sb_fs_annually_other_amount].present?
+          post[:sb_fs_annually_other_amount] =
+            options[:sb_fs_annually_other_amount]
+        end
```

Comment:
Could do something like

```ruby
post.update(options.slice(
  :sb_fs_quarterly_amount,
  :sb_fs_quarterly_other_amount,
  ...
).compact)
```

You could have the same behavior with `.delete_if { |_, v| !v.present? }` but since these are booleans or numbers, that's probably not required?